libs/common/Networking/TcpNetworkHandlerBase.cs

```diff
@@ -135,31 +135,58 @@ void Dispose(SocketAsyncEventArgs e)
         }
 
         void RecvEventArg_Completed(object sender, SocketAsyncEventArgs e)
+        {
+            if (e.BytesTransferred == 0 || e.SocketError != SocketError.Success || serverHook.Disposed)
+            {
+                // No more things to receive
+                Dispose(e);
+                return;
+            }
+            var networkBufferProcessingTask = OnNetworkReceiveAsync(e.BytesTransferred);
+            if (networkBufferProcessingTask.IsCompletedSuccessfully)
+            {
+                NetworkBufferProcessed(networkBufferProcessingTask, sender, e);
+                return;
+            }
+            networkBufferProcessingTask.ContinueWith((res) =>
```

Comment:
This `ContinueWith` means RecvEventArg_Completed will return before processing is complete. What if the next packet arrives immediately and gets processed in parallel for the same session?

Reply:
We do e.AcceptSocket.ReceiveAsync only in ContinueWith on success, so for same socket/session it shouldn't be a problem? I haven't seen it failing or throw any errors yet either